### ISSUE TITLE
Restic fails to restore volume if corresponding pod doesn't exist in cluster but deployment does

### ISSUE THREAD

**Issue by rnarenpujari** [vmware-tanzu/velero#5242]
Restore partially fails with
```
time="2022-08-22T13:12:54Z" level=error msg="unable to successfully complete restic restores of pod's volumes" error="pod volume restore failed: error running restic restore, cmd=restic restore --repo=azure:azure-blob:/kind/restic/default --password-file=/tmp/credentials/velero/velero-restic-credentials-repository-password --cache-dir=/scratch/.cache/restic 31ae352d --target=., stdout=, stderr=: chdir /host_pods/d6ad56b2-1e1d-48a5-8b7c-36f9414929dc/volumes/kubernetes.io~empty-dir/persistent-storage: no such file or directory" logSource="pkg/restore/restore.go:1560" restore=velero/bky-1-20220822184238
```
I discovered this while restoring a backup into a different cluster but looks like it can happen in any scenario where the backed up pod doesn't exist on the cluster but its deployment does.
Looks like what's happening is velero tries to restore the volume data ([link](https://github.com/vmware-tanzu/velero/blob/082d680d7b0cbf1127d3dd2d29e5c706ab88232e/pkg/restore/restore.go#L1398)) into the pod just created by restore ([link](https://github.com/vmware-tanzu/velero/blob/082d680d7b0cbf1127d3dd2d29e5c706ab88232e/pkg/restore/restore.go#L1246)). But since the deployment already has sufficient replicas (or maybe because it's not in the current replica set), the created pod gets deleted almost immediately, and so the destination path (using the pod uid) doesn't exist during the restic restore. So what's weird is that a) the pod gets restored and deleted immediately and b) the overall restore partially fails.  

**What steps did you take and what happened:**
Can easily replicate this using a simple deployment: [dep.txt](https://github.com/vmware-tanzu/velero/files/9399881/dep.txt)
```
% velero --kubeconfig ~/.kube/tmp1 install \
    --provider azure \
    --plugins velero/velero-plugin-for-microsoft-azure:v1.5.0 \
    --bucket <snip> \
    --secret-file ./credentials-velero \
    --backup-location-config resourceGroup=<snip>,storageAccount=<snip> \
    --prefix kind \
    --use-restic
```
```
% k --kubeconfig ~/.kube/config apply -f dep.yaml
```
```
% velero --kubeconfig ~/.kube/tmp1 backup create bck-1 --include-namespaces default --default-volumes-to-restic
```
```
% k --kubeconfig ~/.kube/tmp1 rollout restart deployment app
```
```
% velero --kubeconfig ~/.kube/tmp1 restore create --from-backup bck-1
```
Reproduced this using both emptyDir and PVC and on both velero 1.9.1 and 1.8.1.

**What did you expect to happen:**
Seems like too common a scenario to cause the overall restore to (partially) fail?

**The following information will help us better understand what's going on**:

_If you are using velero v1.7.0+:_  
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 
Two different repros:
[bundle-2022-08-22-18-43-44.tar.gz](https://github.com/vmware-tanzu/velero/files/9400030/bundle-2022-08-22-18-43-44.tar.gz)
[bundle-2022-08-23-11-18-11.tar.gz](https://github.com/vmware-tanzu/velero/files/9400034/bundle-2022-08-23-11-18-11.tar.gz)

**Anything else you would like to add:**
NA

**Environment:**

- Velero version (use `velero version`): 1.9.1
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`): 1.22,1.23,1.24
- Kubernetes installer & version: Kind, EKS
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by blackpiglet** [comment]
I agree Restic cannot work well in this situation, but by current design, it's not easy to handle this case. Velero actually breaks down the relateionship between controller and its resources, e.g. pods, RS and deployment. Velero will backup and restore pods first, then the RS, at last is deployment, and there is no mechism to check whether the pods are deleted or not after successfully creation.
PVB and PVR are handled by Restic DaemonSet, maybe we can add some logic in PVB and PVR controller to check whether pods exist before starting Restic commads.

**Comment by rnarenpujari** [comment]
> PVB and PVR are handled by Restic DaemonSet, maybe we can add some logic in PVB and PVR controller to check whether pods exist before starting Restic commads.  

Looks like the path is checked shortly before running the restic command which I guess means the pod would have been running at the time: https://github.com/vmware-tanzu/velero/blob/e8494418d4135f0b26be5522b1bb9317c222c3c3/pkg/controller/pod_volume_restore_controller.go#L235
So perhaps the pod was deleted while the restic restore was going on.

**Comment by blackpiglet** [comment]
I agree. Before running Resic command, Pod volume restore controller will check whether pod's `restic-wait` InitContainer is running, so I suppose the pod is deleted after Restic command started.

https://github.com/vmware-tanzu/velero/blob/main/pkg/controller/pod_volume_restore_controller.go#L208-L214

**Comment by reasonerjt** [comment]
This problem happens when a user tries to use velero to restore pod to a ns where a deployment existing managing the pods match the selector, and there will be a race-condition that the deployment will kill a pod restored by velero.

This seems a corner case b/c the user normally will restore the workload where there's no such pre-existing deployment.

A reasonable fix may be to skip the pods during the restore if they are managed by some pre-exisitng deployments.

**Comment by sseago** [comment]
@reasonerjt the problem with skipping pods if they're managed by deployments is that then restic/kopia restore will never succeed, since we won't have the initcontainer, etc. on the pods in the cluster.
